expression/expression.go

```diff
@@ -145,6 +145,34 @@ func ComposeDNFCondition(ctx context.Context, conditions ...Expression) Expressi
 	return composeConditionWithBinaryOp(ctx, conditions, ast.LogicOr)
 }
 
+// ExtractDNFConditions extract DNF expression's leaf item.
+// e.g. or(or(a=1, a=2), or(a=3, a=4)), we'll get [a=1, a=2, a=3, a=4].
+func ExtractDNFConditions(DNFCondition *ScalarFunction) []Expression {
+	var ret []Expression
+	for _, arg := range DNFCondition.GetArgs() {
+		if sf, ok := arg.(*ScalarFunction); ok && sf.FuncName.L == ast.LogicOr {
+			ret = append(ret, ExtractDNFConditions(sf)...)
+		} else {
+			ret = append(ret, arg)
+		}
+	}
+	return ret
+}
+
+// ExtractCNFConditions extract CNF expression's leaf item.
+// e.g. and(and(a>1, a>2), and(a>3, a>4)), we'll get [a>1, a>2, a>3, a>4].
+func ExtractCNFConditions(CNFCondition *ScalarFunction) []Expression {
```

Comment:
Seems like that this function is almost the same with `ExtractDNFConditions`, could we merge them?